grpc.go

```diff
@@ -515,6 +515,34 @@ func (a *agentGRPC) CreateContainer(ctx context.Context, req *pb.CreateContainer
 		return emptyResp, err
 	}
 
+	for idx, ociMnt := range ociSpec.Mounts {
+		splitSourceSlice := strings.Split(ociMnt.Source, "/")
+		if len(splitSourceSlice) > 1 {
+			k8sStorageType := splitSourceSlice[len(splitSourceSlice)-2]
+			if k8sStorageType == "emphemeral" {
```

Comment:
- Is this a reliable test? This appears to be looking for the word "emphemeral" in the `ociMnt.Source` *path* (and that "magic tag" might change at some point.
- This block uses deep indentation. You could save one level by restructuring slightly:
  ```
  if k8sStorageType != "emphemeral" {
      continue
  }

  ...
  ```

Reply:
@jodh-intel Is it very reliable to check for `k8sStorageType == "emphemeral"` condition. The string `emphemeral` is introduced by this patch, https://github.com/kata-containers/runtime/pull/307/files#diff-8a56f5a1fa5971cc43e650ffa24f1e2bR376 

That patch in `runtime` loops over the given OCI mounts to look for the mounts used by k8s for ephemeral storage. It then replaces those mount paths on the host with the mount paths in the guest by adding `"/ephemeral"` to the path. So it's very much in our control the existance of that string. Right now without this patch, an ephemeral volume is passed to guested via 9p. 

e.g. A typical `ephemeral volume` path looks like, `/var/lib/kubelet/pods/366c3a75-4869-11e8-b479-507b9ddd5ce4/volumes/kubernetes.io~empty-dir/cache-volume`

But this resides on the host. We want that volume to be provisioned inside our VM. We can keep the same path or we can make it simpler since we have the total control on what happens inside the sandbox VM. So I replace that original source path with much simpler `"/ephemeral"`. Inside the VM that above path becomes, `/ephemeral/cache-volume`

Reply:
@jodh-intel I will add some unit test cases.